### ISSUE TITLE
WebRTC "leak" still occurs and setting to prevent it has been deliberately removed

### ISSUE THREAD

**Issue by meyergru** [uBlockOrigin/uBlock-issues#2267]
### Prerequisites

- [X] I verified that this is not a filter issue (MUST be reported at [filter issue tracker](https://github.com/uBlockOrigin/uAssets/issues))
- [X] This is not a [support issue or a question](https://old.reddit.com/r/uBlockOrigin/)
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uBlock-issues/issues?q=is%3Aissue) to avoid opening a duplicate issue
- [X] The issue is not present after wholly disabling uBlock Origin ("uBO") in the browser
- [X] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I report is not a normal behavior

### I tried to reproduce the issue when...

- [X] uBO is the only extension
- [X] uBO with default lists/settings
- [X] using a new, unmodified browser profile

### Description

I know that the WebRTC leakage fix option has been removed a while ago because it apparently has been fixed in the browsers already and it supposedly creates problems if it is applied.

Therefore, the old issue https://github.com/uBlockOrigin/uBlock-issues/issues/1723 has been closed to the public.

I also admit that I had to check the box above stating "The issue is not present after wholly disabling uBlock Origin ("uBO") in the browser" just in order to be able to file this bug report while the statement itself is actually untrue.

However, there still is a privacy concern in how this works which I found when I tried WebRTC leak tests over an anonymizing proxy: Using that, one should assume that the originating IP is being hidden from the website. This is true for the HTTP request, but not for the WebRTC IP. Even when using a proxy, WebRTC shows the real IP (i.e. the NATed IP, not the one on the LAN).

Thus, using WebRTC, the cloaking via anonymous proxies can be circumvented, which IS a privacy issue.

I had not followed the discussion and only found out by chance. I was quite disappointed to learn that a. most browsers allow WebRTC by default (even Brave does it!) and b. uBlock Origin does not prevent it.

### A specific URL where the issue occurs

https://hide.me/de/webrtc-leak-test

### Steps to Reproduce

1. While you have access to the internet using public IP X, use an anonymizing proxy somewhere on the internet with IP Y.
2. Browse to a website that shows your WebRTC IP (most of those found via searching for "webrtc leak test").
3. You will see that IP X is visible on that website.

### Expected behavior

I do not expect to have my public IP disclosed despite using a proxy.

### Actual behavior

My public IP was disclosed.

### uBlock Origin version

1.44.2

### Browser name and version

Firefox 104.0.2

### Operating System and version

Windows 10

**Comment by gwarser** [comment]
Which one proxy you are using and how it is configured?

**Comment by meyergru** [comment]
It's a squid proxy, anonymizing by setting:

`
forwarded_for delete

via off
`

**Comment by gwarser** [comment]
I mean, how you are sure your browser is using it?

**Comment by Yuki2718** [comment]
What's the matter? The removed setting which just turn browser's option on was to prevent leakage of your private IP address, not public address you discuss here. Hiding that is simply not a job of a browser extension.

**Comment by Yuki2718** [comment]
> It's a squid proxy, anonymizing by setting:
>

squid is a http proxy, how this can hide your public IP via WebRTC (UDP)?

**Comment by meyergru** [comment]
I think that is highly debatable. What else does this extension do that to supply privacy mechanisms the browsers do not offer themselves.

Considering the reason to remove the setting (namely that the browsers now fix the local IP leakage themselves), it seems that the very reason to have it was indeed to hide information in the first place.

I argue that the leakage has only partly been removed, so there is less ground to remove the setting. Essentially, you are providing a different reason to remove it (that I do not buy).

Regarding your second argument: using squid to hide your identity is a purge neccessity in a lot of countries where free speech does not exist. As I said, I was very disappointed to learn that WebRTC was introduced without a proper "lark's vomit" warning, essentially undermining that. So, technically you are correct in saying that a proxy does not hide your public IP via WebRTC. However, without WebRTC, an HTTP proxy perfectly hides your public IP (modulo DNS leaks).

**Comment by gorhill** [comment]
The browsers are not supposed to leak these, this was fixed years ago. I suggest you rather open an issue with whichever browser you found suffer the issue, I do not consider a uBO issue.

**Comment by meyergru** [comment]
That is exactly what I am saying: most browsers leak THIS information - I have verified for Edge, Chrome, Firefox and even Brave. 

What has been fixed is the LAN IP and the IPv6 information leakage, not that the public IPv4 is still visible per default.

So, it is correct that this should be fixed in (all) the browser(s), As for uBo, it is not really a "bug", but an incompleteness. As I wrote: Why was the setting there in the first place? To fix something that the browsers didn't. The assumption that they now do this themselves is partly wrong.

**Comment by gorhill** [comment]
> What has been fixed is the LAN IP and the IPv6 information leakage

In my tests, I could not see my ISP IP addresses when behind a VPN -- when using a system wide VPN (Firefox VPN).

**Comment by meyergru** [comment]
Correct, because in that case, the wohle network stack is diverted to the VPN endpoint. You will see it when you use an anonymizing proxy, though.

**Comment by meyergru** [comment]
I might as well disable WebRTC in all my browsers - although this is tedious because they have have different means of enabling that, unless I use a specialised extension to do it. Probably not uBo, even if it once did that job.

**Comment by Yuki2718** [comment]
Well, the leak won't occur if system-wide VPN is used or a proxy on top of browser *plus* WebRTC IP handling policy set to "Disable non-proxied UDP". Apparently only Brave expose this option to user, at least under GUI.

**Comment by Yuki2718** [comment]
Related Brave issue: https://github.com/brave/brave-browser/issues/22515

**Comment by meyergru** [comment]
No, I just set the Firefox proxy to a server that runs squid with the settings I gave. You can do that with an extension like proxy-switch, manually or via WPAD.

**Comment by Yuki2718** [comment]
What matters here is if you use a simple proxy but not a full-tunnel VPN, your ISP IP will be exposed by WebRTC, which can be fixed if uBO sets WebRTC IP handling policy to "Disable non-proxied UDP". Ofc there's no guarantee a proxy totally hides your IP (those who want that should use VPN). But this is something to consider about.

**Comment by krystian3w** [comment]
Old version of add-on hide proxy leak?

**Comment by gorhill** [comment]
It's the same uBO package on both desktop or android, the setting is just ignored on desktop.
